File: truevfs-ext/truevfs-ext-logging/src/main/scala/net/java/truevfs/ext/logging/LogBufferPoolDecorator.scala
/*
 * Copyright (C) 2005-2012 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.logging

import net.java.truecommons.cio._
import net.java.truevfs.kernel.spec.spi._

/**
  * @author Christian Schlichtherle
  */
@deprecated("This class is reserved for exclusive use by the [[net.java.truevfs.kernel.spec.sl.IoBufferPoolLocator.SINGLETON]]!", "1")
final class LogBufferPoolDecorator extends IoBufferPoolDecorator with Immutable {

  override def apply(pool: IoBufferPool): IoBufferPool =
    LogMediator.instrument(pool)

  /** Returns -100. */
  override def getPriority = -100
}

File: truevfs-ext/truevfs-ext-logging/src/main/java/net/java/truevfs/ext/logging/LogBufferPoolDecorator.java
/*
 * Copyright © 2005 - 2021 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.logging;

import global.namespace.service.wight.annotation.ServiceImplementation;
import net.java.truecommons.cio.IoBufferPool;
import net.java.truevfs.kernel.spec.spi.IoBufferPoolDecorator;

/**
 * @author Christian Schlichtherle
 */
@ServiceImplementation(priority = -300)
public final class LogBufferPoolDecorator implements IoBufferPoolDecorator {

    @Override
    public IoBufferPool apply(IoBufferPool pool) {
        return LogMediator.SINGLETON.instrument(pool);
    }
}
